### PR TITLE
Added skiping `FileTarget::class` with message its `empty` array.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | https://github.com/yiisoft/yii2/pull/16208

### CODE REVIEW

**Review by samdark — APPROVED** [review]
Looks 👍 Need a line for CHANGELOG.

**Comment by samdark** [comment]
Also, is it the same in Yii3 log package?

File: framework/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ Yii Framework 2 Change Log
 2.0.49 under development
 ------------------------
 
+- Bug #16208: Added skiping `FileTarget::class` with message its empty array (terabytesoftw)
```
Comment (bizley) [inline framework/CHANGELOG.md:7]:
```suggestion
- Bug #16208: Fix `yii\log\FileTarget` to not export empty messages (terabytesoftw)
```

File: framework/log/FileTarget.php [hunk 2]
```diff
@@ -112,6 +112,13 @@ public function export()
         }
 
         $text = implode("\n", array_map([$this, 'formatMessage'], $this->messages)) . "\n";
+
+        $trimmedText = trim($text);
+
+        if (empty($trimmedText)) {
```
Comment (bizley) [inline framework/log/FileTarget.php:118]: Can we move this whole check to the first line? To not create directories if there is nothing to export anyway.
Also - why not use `$trimmedText` later on when actually saving to the file?

**Review by bizley — CHANGES_REQUESTED** [review]

**Review by bizley — APPROVED** [review]

**Review by Arhell — APPROVED** [review]

**Comment by samdark** [comment]
Thanks!

**Pull request merged**